src/preprocess.py

```diff
@@ -97,21 +170,63 @@ def build_inverted_index(
     Returns:
         dict -- inverted index of given type
     """
-    collection = remove_stop_words(collection, stop_words_path)
-    collection = collection_lemmatize(collection)
-    index = {}
-    for key in collection.keys():
-        terms = set(collection[key])
-        for t in terms:
-            if not t in index:
-                index[t] = []
-            if type_index == 1:
-                index[t].append(key)
-            elif type_index == 2:
-                index[t].append((key,collection[key].count(t)))
-            elif type_index == 3:
-                index[t].append((key,tuple([index for index in range(len(collection[key])) if collection[key][index] == t])))
-    return index
+    collection = tokenize_collection(collection)
+    collection = remove_stop_words_collection(collection, stop_words_path)
+    collection = lemmatize_collection(collection)
+    collection = remove_stop_words_collection(collection, stop_words_path)
+    mapping = OrderedDict()
+    index = OrderedDict()
+
+    doc_id = 0
+    if type_index == 1:
```

Comment:
This part could be refactored with auxiliary function for lisibility.

Reply:
todo later